completions/xdg-mime

```diff
@@ -2,9 +2,18 @@
 
 _xdg_mime_mimetype()
 {
-    COMPREPLY+=($(compgen -S / -W 'application audio font image message model
-                                    multipart text video' -- "$cur"))
-    [[ ${COMPREPLY-} == */ ]] && compopt -o nospace
+    local d i
+    for d in /usr/share/mime /usr/local/share/mime; do
+        command cd /usr/share/mime 2>/dev/null || continue
+        trap "command cd - &>/dev/null" RETURN
```

Comment:
I feel the handling of RETURN could be subtle due to the same problem as discussed in https://github.com/scop/bash-completion/issues/786#issuecomment-1221243215.

In theory, we can combine it with trapping SIGINT by saving/restoring the users' SIGINT handler. However, in experimenting it with #739, I found that Bash has a bug that the internal readline state is broken when we use `trap ... INT` inside the completion functions called through `complete -F`. I have sent [a patch](https://lists.gnu.org/archive/html/bug-bash/2022-09/msg00005.html) to Bash, but Bash 4.3..5.2 are affected at least, and these older versions will not be fixed.

Instead, I would propose a new option `-C dir` of `_filedir`. `cd $dir` may be performed in the subshell of `_filedir` as `toks=($(cd "$dir" && compgen ...))`.

Reply:
Damn, I had completely forgotten about that. That's a bummer, and would be great to have something to remind, because I'm bound to forget about this again (it feels so natural when one works with Go and its `defer`).

`_filedir -C` makes sense though; we have a few completions that could make use of it.

(Aside, I kind of wouldn't want to introduce more reliance on `_filedir` but rather to get rid of it some day, but the day where that would be feasible might never come...)

Reply:
#847 has some `_filedir -C` experimentation.

Reply:
4fc5c89a462d22fdc685af79c8eade8ae476dba4 switches to an approach not involving `_filedir` or `trap`, lifting draft.